### DISCUSSION TITLE
Hide "Write a reply" textboxes

### DISCUSSION THREAD

**Discussion by alkisg** [community/community#3137] [Discussions]
Let's see an example: https://github.com/ltsp/ltsp/discussions/361#discussioncomment-268741

There, a user replied "inline" to my [previous message](https://github.com/ltsp/ltsp/discussions/361#discussioncomment-265208). 2 days ago, [he had also replied](https://github.com/ltsp/ltsp/discussions/361#discussioncomment-265243) to the bottom of the discussion. This makes it difficult for me to locate the new replies in this discussion.

So I'm asking to "advertise" inline/threaded discussions **less** in the github UI, so that they're only used when there's really need for them. One way to do that would be to replace all "Write a reply" textboxes with "Reply in thread" buttons, that will appear right next to the "Mark as answer" buttons, [like slack does it](https://www.atlassian.com/blog/halp/how-and-why-to-use-slacks-threaded-messages-in-channels-and-apps).

This will also simplify the UI a lot, now it's rather loaded with all the "Write a reply" inline boxes. But the most important part is that now it's very difficult to read the discussions sequentially.

**Comment by vanbroup** [comment]
I agree that users can get confused, having reply on comments is great and keeps the relevent comments in focus.

- Instead of `Write a reply` it could state `Reply to ...`, `Reply to comment from ....` or `Reply to answer from ....`
- `Suggest an answer` could be a button, like the `New conversation` button in Teams. Which was changed to a button for the same reason.

**Comment by evi-liu** [comment]
Thank you for this feedback. We are currently iterating on our design for comments and threading, and your feedback has been noted.

**Comment by jcw** [comment]
I've started a project with discussions enabled, and right away the confusion sets in. In issues, everything is linear, where my first instinct is to scroll to the bottom and, if a reply is to be made, to type it right there. In discussions, especially if up/down-voting is to be meaningful, the reply-at-the-bottom convention is probably exactly the _worst_ one to use.

My suggestion would be to somehow very clearly indicate in the bottom reply area that it is probably not the best place to add a note. Perhaps the whole discussion page should have a very distinct different style from issues. Or the bottom reply area could start off in a collapsed mode, with some clear wording that this is not really a "reply" but a new note alongside all the other (threaded) notes.

Perhaps this can be solved with proper terminology. A "conversation" (already) is the whole page. A "thread" can then be a sequence of comments, each very much associated with the text just above it (and nesting/indenting, up to a point, maybe even just 2 deep). The box at the bottom would then start a new thread, which should probably often be made less visible and convenient than adding a note to an existing thread. As I said before, a collapsed area might help. Maybe the last box on the last thread should be the "big" box, which then adds to the thread instead of starting a new one.

Even for this reply, I ended up using the box at the bottom (i.e. a new thread). Mostly because I thought that a long message like this one is perhaps best _not_ inserted in a the middle of the discussion so far. So there's also the tension between "barging in" and adding a somewhat longer message at the end ... which also affects which of the many boxes to start typing in.

Would it be useful to only present a _single_ comment box, and then decide _later_ where to place it (as writer, or as maintainer)?

As for the issue of reading things back once several threads are active in the conversation - that's a whole different can of worms.

(Note: I'm not a UX designer, and I've only just started using GH discussions)

**Comment by tve** [comment]
Is this supposed to be a discussion or a Q&A? If it's a discussion there's a certain linearity implied and all the "write a reply" under each comment and voting up/down to change the order seems out of place. If it's a Q&A and each "comment" really is more of an answer (e.g. stackoverflow style) then that's great, but maybe then the whole feature should be called "Q&A" and not "discussion" and other wording should be adjusted accordingly? Or possibly each category could be either a (linear) discussion category or it could be a Q&A category (that would be a nice model IMHO).
